cloud/src/meta-service/meta_service_resource.cpp

```diff
@@ -509,6 +509,80 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
     LOG(INFO) << vault_msg;
 }
 
+static int alter_storage_vault(InstanceInfoPB& instance, std::unique_ptr<Transaction> txn,
+                               const StorageVaultPB& vault, MetaServiceCode& code,
+                               std::string& msg) {
+    if (!vault.has_obj_info()) {
+        code = MetaServiceCode::INVALID_ARGUMENT;
+        std::stringstream ss;
+        ss << "Only s3 vault can be altered";
```

Comment:
Why not HDFS?

Reply:
Currently i'm not sure which property of hdfs vault can be altered. As the pr title says, this pr is for `S3 Vault`.